src/shared/ui/Typography/Typhgraphy.tsx

```diff
@@ -0,0 +1,98 @@
+import { ComponentPropsWithoutRef } from 'react';
+import { cva } from 'class-variance-authority';
+
+export type TypographyVariant =
+  | 'Title1'
+  | 'Title2'
+  | 'Title3'
+  | 'Title4'
+  | 'Subtitle1'
+  | 'Subtitle2'
+  | 'Subtitle3'
+  | 'Body1'
+  | 'Body2'
+  | 'Body3'
+  | 'Caption1'
+  | 'Caption2'
+  | 'ButtonTitle1'
+  | 'ButtonTitle2';
+
+export type Props = {
+  weight?: 'light' | 'normal' | 'medium' | 'semibold' | 'bold';
+  variant: TypographyVariant;
+  className?: string;
+} & ComponentPropsWithoutRef<'p'>;
+
+export const variantClasses = cva('whitespace-pre-wrap', {
+  variants: {
+    type: {
+      Title1: 'md:text-4xl text-2xl font-bold',
+      Title2: 'md:text-3xl text-xl font-bold',
+      Title3: 'md:text-2xl text-xl font-bold',
+      Title4: 'md:text-xl text-lg font-semibold',
+      Subtitle1: 'md:text-xl text-lg font-semibold',
+      Subtitle2: 'md:text-lg text-base font-semibold',
+      Subtitle3: 'md:text-base text-base font-semibold',
+      Body1: 'md:text-xl text-lg font-medium',
+      Body2: 'md:text-lg text-base font-medium',
+      Body3: 'md:text-base text-base font-medium',
+      Caption1: 'md:text-base text-base font-normal',
+      Caption2: 'md:text-sm text-sm font-normal',
+      ButtonTitle1: 'md:text-xl text-base font-bold',
+      ButtonTitle2: 'md:text-base text-base font-bold',
+    },
+    weight: {
+      light: 'font-light',
+      normal: 'font-normal',
+      medium: 'font-medium',
+      semibold: 'font-semibold',
+      bold: 'font-bold',
+    },
+  },
+  defaultVariants: {
+    type: 'Body1',
+    weight: 'medium',
+  },
+});
+
+export function Typography({
+  children,
+  weight = 'medium',
+  variant = 'Body1',
+  className = '',
+  ...props
+}: Props) {
+  const baseClasses = variantClasses({ type: variant, weight });
+  const combinedClasses = `${baseClasses}${className ? ` ${className}` : ''}`;
+
+  return (
+    <p className={combinedClasses} {...props}>
```

Comment:
다음과 같이 코드를 줄일 수 있을 것 같아요.
```suggestion
  return (
    <p className={cn(variantClasses({ type: variant, weight }), className)} {...props}>
```

Reply:
25e11a9aef327341c145f5fbd1ca945da8a5be4b
수정하였습니다 감사합니다 !